yt_dlp/extractor/picarto.py

```diff
@@ -102,21 +111,30 @@ class PicartoVodIE(InfoExtractor):
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        webpage = self._download_webpage(url, video_id)
+        data = self._download_json(
+            'https://ptvintern.picarto.tv/ptvapi', video_id, query={
+                'query': '''{
+  video(id: "%s") {
+    id
+    title
+    file_name
+    video_recording_image_url
+    channel {
+      name
+    }
+  }
+}''' % video_id,
+            })['data']['video']
 
-        vod_info = self._parse_json(
-            self._search_regex(
-                r'(?s)#vod-player["\']\s*,\s*(\{.+?\})\s*\)', webpage,
-                'vod player'),
-            video_id, transform_source=js_to_json)
+        title = data['file_name']
+        netloc = urllib.parse.urlparse(data['video_recording_image_url']).netloc
 
         formats = self._extract_m3u8_formats(
-            vod_info['vod'], video_id, 'mp4', entry_protocol='m3u8_native',
-            m3u8_id='hls')
+            f'https://{netloc}/stream/hls/{title}/index.m3u8', video_id, 'mp4', m3u8_id='hls')
 
         return {
             'id': video_id,
-            'title': video_id,
-            'thumbnail': vod_info.get('vodThumb'),
+            'title': title,
```

Comment:
Why not `data.get('title')`?

Reply:
`title` is used in the m3u8 url

Reply:
ah nvm I see what you mean

Reply:
Oh, at first I was concerned that vods with the same title could overwrite each other, but I see now that the video id is appended to the filename so that would never happen. Will update that.